Use buttons for buttons, a instead of span and add aria-label

arielj opened the pull request:
This PR improves the accessibility of the app by providing better support for keyboard navigation with better focus targets, focus indicators and labels:
- Added an SvgButton to wrap the different svg tags and font awesome icons
- Replaced many `span.link` to be `a` tags with `href="#"` so they can be focused when tabbing
- Added aria-labels to the ToggleSwitch element so a screen reader can match what's each switch for
- I also fixed the focus indication for game cards in the grid layout

I tried the different screens and I was able to change settings, install/uninstall a game, start it, etc.

This closes #745 

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Tested the feature and it's working on a current and clean install.
- [x] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

flavioislima commented:
Nice!
I'll try to review it today or tomorrow.
Btw, about the gamecard, I noticed you haven't changed it from a div to something else. Don't worry about that because I fixed it on my #802 PR.
I removed the `div` and adapted the `Link` so it can be browsed with the keyboard.

arielj commented:
> Nice! I'll try to review it today or tomorrow. Btw, about the gamecard, I noticed you haven't changed it from a div to something else. Don't worry about that because I fixed it on my #802 PR. I removed the `div` and adapted the `Link` so it can be browsed with the keyboard.

oh great! yeah I just fixed that one with CSS to not break anything (using the focus-within selector), but it's better if it's a link tag instead

flavioislima commented on src/components/UI/Header/index.css:
```diff
@@ -129,15 +128,22 @@ option {
 
-.headerIcons > *:hover {
+.headerIcons > .svg-button {
+  color: white;
```
We should always use variables for the colors.
```suggestion
  color: var(--text-primary);
```

flavioislima commented on src/screens/Library/components/GameCard/index.tsx:
```diff
@@ -141,11 +142,9 @@ const GameCard = ({
-          onClick={() => handleUpdate()}
-        />
+        <SvgButton onClick={() => handleUpdate()}>
+          <FontAwesomeIcon size={'2x'} icon={faRepeat} />
```
When I tested this branch, for some reason this icon is Black instead of green. I think some class is overriding it.
so try to make it use the correct color.

arielj replied:
fixed this, thanks

flavioislima reviewed (changes requested):
Overall looks pretty good, left a few minor comments though.

flavioislima commented:
> > Nice! I'll try to review it today or tomorrow. Btw, about the gamecard, I noticed you haven't changed it from a div to something else. Don't worry about that because I fixed it on my #802 PR. I removed the `div` and adapted the `Link` so it can be browsed with the keyboard.
> 
> oh great! yeah I just fixed that one with CSS to not break anything (using the focus-within selector), but it's better if it's a link tag instead

After releasing the new beta I noticed that my solution doesn't work as expected because then we cannot click on the icons. Like, since the link is the wrapper, it has preference over the rest of the elements. So I'll revert it and we will use yours :)

arielj commented:
> > > Nice! I'll try to review it today or tomorrow. Btw, about the gamecard, I noticed you haven't changed it from a div to something else. Don't worry about that because I fixed it on my #802 PR. I removed the `div` and adapted the `Link` so it can be browsed with the keyboard.
> > 
> > 
> > oh great! yeah I just fixed that one with CSS to not break anything (using the focus-within selector), but it's better if it's a link tag instead
> 
> After releasing the new beta I noticed that my solution doesn't work as expected because then we cannot click on the icons. Like, since the link is the wrapper, it has preference over the rest of the elements. So I'll revert it and we will use yours :)

oh sure, I remove the css, then read this message and now I re-added the css

arielj commented:
@flavioislima done, just merged main into this branch and fixed the css

dragonDScript reviewed (commented):
Thanks for the effort, and for caring about accesibility. Would set user-select to none worsen accesibility? This PR LGTM but I'll re-review later

flavioislima reviewed (approved):
Nice, it's working perfectly now! thanks! :D
